server/src/main/java/io/deephaven/server/plugin/python/PyObjectRefCountedNode.java

```diff
@@ -0,0 +1,40 @@
+package io.deephaven.server.plugin.python;
+
+import io.deephaven.engine.liveness.LivenessArtifact;
+import io.deephaven.engine.liveness.ReferenceCountedLivenessNode;
+import io.deephaven.util.annotations.ScriptApi;
+import org.jpy.PyObject;
+
+/**
+ * Provides a mapping between Python refcount and Deephaven's liveness mechanism, allowing liveness scopes to manage the
+ * single <a href='https://docs.python.org/3/glossary.html#term-strong-reference'>strong reference</a> that the PyObject
+ * instance represents. This way, as long as PyObjectRefCountedNode instances are managed correctly by their parent
+ * scope, the PyObject strong reference will be correctly released when the object is no longer used.
+ */
+public class PyObjectRefCountedNode extends LivenessArtifact {
```

Comment:
So, I think `PyObject`s _should_ be correctly released already. I don't know about the tie-ins to the liveness mechanism and if they are necessary for correctness or not...

Reply:
Right now PyObject isn't instantiate-able, so this could fill that hole. But would making PyObject constructors public achieve almost the same goal besides the liveness benefits? Just a question.

Reply:
I don't think that would help, even ignoring the liveness issue - the problem of #1775 is that we can't freely pass both java and python objects from python to java - we need some "real" java object that contains the PyObject.